pkg/ring/replication_strategy.go

```diff
@@ -0,0 +1,55 @@
+package ring
+
+import (
+	"fmt"
+	"time"
+)
+
+func (r *Ring) replicationStrategy(ingesters []*IngesterDesc) (
```

Comment:
please explain what the overall purpose of this function is, and what the `ingesters` parameter should contain